app/src/js/components/Pdr/pdr.js

```diff
@@ -39,247 +39,238 @@ import ErrorReport from '../Errors/report';
-              PDR: {pdrName}
-            </h1>
-            <AsyncCommand
-              action={this.deletePdr}
-              success={this.navigateBack}
-              status={deleteStatus}
-              className={'form-group__element--right'}
-              confirmAction={true}
-              confirmText={deleteText(pdrName)}
-              text={deleteStatus === 'success' ? 'Deleted!' : 'Delete'}
-            />
-            {lastUpdated(queriedAt)}
-            {this.renderProgress(record)}
-            {error && <ErrorReport report={error} />}
-          </div>
-        </section>
+  const granulesCount = get(record, 'data.stats', []);
+  const granuleStatus = Object.keys(granulesCount).map((key) => ({
+    count: granulesCount[key],
+    key: (key === 'processing') ? 'running' : key
+  }));
 
-        <section className="page__section">
-          <div className="heading__wrapper--border">
-            <h2 className="heading--medium with-description">PDR Overview</h2>
-          </div>
-          <Metadata data={record.data} accessors={metaAccessors} />
-        </section>
+  const metaAccessors = [
+    {
+      label: 'Provider',
+      property: 'provider',
+      accessor: (d) => (
+        <Link
+          to={{
+            pathname: `/providers/provider/${d}`,
+            search: getPersistentQueryParams(location),
+          }}
+        >
+          {d}
+        </Link>
+      ),
+    },
+    {
+      label: strings.collection,
+      property: 'collectionId',
+      accessor: collectionLink,
+    },
+    {
+      label: 'Execution',
+      property: 'execution',
+      accessor: (d) => (d
+        ? (
+        <Link
+          to={{
+            pathname: `/executions/execution/${path.basename(d)}`,
+            search: getPersistentQueryParams(location),
+          }}
+        >
+          link
+        </Link>
+          )
+        : (
+            nullValue
+          )),
+    },
+    {
+      label: 'Status',
+      property: 'status',
+      accessor: displayCase,
+    },
+    {
+      label: 'Timestamp',
+      property: 'timestamp',
+      accessor: fullDate,
+    },
+    {
+      label: 'Created at',
+      property: 'createdAt',
+      accessor: fullDate,
+    },
+    {
+      label: 'Duration',
+      property: 'duration',
+      accessor: seconds,
+    },
+    {
+      label: 'PAN Sent',
+      property: 'PANSent',
+      accessor: bool,
+    },
+    {
+      label: 'PAN Message',
+      property: 'PANmessage',
+    },
+  ];
 
-        <section className="page__section">
-          <div className="heading__wrapper--border">
-            <h2 className="heading--medium heading--shared-content with-description">
-              {strings.granules}{' '}
-              <span className="num-title">
-                {!Number.isNaN(+count) ? `(${count})` : 0}
-              </span>
-            </h2>
-          </div>
-          <div>
-            <GranulesProgress granules={granuleStatus} />
-          </div>
+  return (
```

Comment:
We're moving big blocks of code here. Are there any functional changes that need to be reviewed closely? I scanned for anything that jumped out as different but it mostly just looks like a move?

Reply:
Another class component that had to be transitioned to a functional component to work with React Router v6.